src/LayerAnalyser.php

```diff
@@ -34,42 +31,19 @@ public function __construct(
     public function analyse(Configuration $configuration, string $layer): array
     {
         $astMap = $this->astRunner->createAstMapByFiles($this->fileResolver->resolve($configuration), $configuration->getAnalyzer());
-        $dependencyResult = $this->resolver->resolve($astMap);
         $classLikeLayerResolver = $this->classLikeLayerResolverFactory->create($configuration, $astMap);
 
-        /** @var string[] $classLikeNames */
-        $classLikeNames = [];
+        $tokenNames = [];
 
         foreach ($astMap->getAstClassReferences() as $classReference) {
-            $classLikeName = $classReference->getClassLikeName();
-            if ($this->isInLayer($layer, $classLikeName, $classLikeLayerResolver)) {
-                $classLikeNames[] = $classLikeName->toString();
+            $classLikeName = $classReference->getTokenName();
+            if (in_array($layer, $classLikeLayerResolver->getLayersByClassLikeName($classLikeName), true)) {
+                $tokenNames[] = $classLikeName->toString();
             }
         }
 
-        foreach ($dependencyResult->getDependenciesAndInheritDependencies() as $dependency) {
```

Comment:
This `foreach` is entirely unnecessary as it is not possible for anything not already covered by the previous `foreach` to be there.